From bdc7c3e1d3917212033f3314236f1ee0d255a224 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Tue, 17 Feb 2026 12:24:38 +0000
Subject: [PATCH] Skip all OEM unlock prompts for GSI

A GSI requires an unlocked bootloader to boot. Suppress all OEM unlock
related UI in the setup wizard:

- Report the bootloader as locked to hide the welcome screen warning
  banner and skip the relock prompt (OemUnlockActivity)
- Hardcode disableOemUnlockingVisible to false so the finish screen
  "Disable OEM unlocking" checkbox is never shown
- Skip the OemLockManager query in FinishActions so
  disableOemUnlockByUser() is never called

Change-Id: I0000000000000000000000000000000000000010
---
 .../setupwizard/action/FinishActions.kt       | 12 ++------
 .../setupwizard/action/WelcomeActions.kt      | 30 ++++---------------
 .../grapheneos/setupwizard/data/FinishData.kt |  6 ++--
 3 files changed, 9 insertions(+), 39 deletions(-)

diff --git a/java/app/grapheneos/setupwizard/action/FinishActions.kt b/java/app/grapheneos/setupwizard/action/FinishActions.kt
index 785083c..ace027e 100644
--- a/java/app/grapheneos/setupwizard/action/FinishActions.kt
+++ b/java/app/grapheneos/setupwizard/action/FinishActions.kt
@@ -49,16 +49,8 @@ object FinishActions {
     }
 
     private fun refreshOemUnlockStatus() {
-        if (Build.isDebuggable()) {
-            val flag = DebugFlags.getBool("isOemUnlockAllowedByUser_override")
-            if (flag != null) {
-                FinishData.oemUnlockingEnabled.value = flag
-                return
-            }
-        }
-
-        FinishData.oemUnlockingEnabled.value =
-            getOemLockManager()?.isOemUnlockAllowedByUser ?: false
+        // GSI requires an unlocked bootloader; never offer to disable OEM unlocking
+        FinishData.oemUnlockingEnabled.value = false
     }
 
     private fun disableOemUnlockByUser() {
diff --git a/java/app/grapheneos/setupwizard/action/WelcomeActions.kt b/java/app/grapheneos/setupwizard/action/WelcomeActions.kt
index bccfa44..ba4fff9 100644
--- a/java/app/grapheneos/setupwizard/action/WelcomeActions.kt
+++ b/java/app/grapheneos/setupwizard/action/WelcomeActions.kt
@@ -116,33 +116,13 @@ object WelcomeActions {
     }
 
     fun next(activity: Activity) {
-        if (Build.isDebuggable() && DebugFlags.getBool("enableUnlockedBootloaderHandling") != true) {
-            // we allow free pass for development features on debug builds of the OS
-            SetupWizard.next(activity)
-            return
-        }
-        if (SetupWizard.isSecondaryUser) {
-            // secondary users should not be bothered for this
-            // the device setup (primary user setup) is already done at this point
-            SetupWizard.next(activity)
-            return
-        }
-        if (WelcomeData.oemUnlocked.value == true) {
-            SetupWizard.startActivity(activity, OemUnlockActivity::class.java)
-        } else {
-            SetupWizard.next(activity)
-        }
+        // GSI requires an unlocked bootloader; skip relock prompt entirely
+        SetupWizard.next(activity)
     }
 
     private fun refreshOemUnlockStatus() {
-        if (Build.isDebuggable()) {
-            val flag = DebugFlags.getBool("isDeviceOemUnlocked_override")
-            if (flag != null) {
-                WelcomeData.oemUnlocked.value = flag
-                return
-            }
-        }
-
-        WelcomeData.oemUnlocked.value = getOemLockManager()?.isDeviceOemUnlocked ?: false
+        // GSI always runs with an unlocked bootloader; report as locked to
+        // suppress the warning banner and disable OEM unlocking prompts
+        WelcomeData.oemUnlocked.value = false
     }
 }
diff --git a/java/app/grapheneos/setupwizard/data/FinishData.kt b/java/app/grapheneos/setupwizard/data/FinishData.kt
index b2b8d84..7b72b9c 100644
--- a/java/app/grapheneos/setupwizard/data/FinishData.kt
+++ b/java/app/grapheneos/setupwizard/data/FinishData.kt
@@ -12,10 +12,8 @@ object FinishData : ViewModel() {
     // whether the "Disable OEM Unlocking" checkbox should be checked by default
     val disableOemUnlockingChecked = !Build.isDebuggable()
 
-    // whether "Disable OEM Unlocking" checkbox should be visible
-    //     visible if bootloader is locked and user is primary
-    val disableOemUnlockingVisible =
-        WelcomeData.oemUnlocked.value?.not() ?: false && SetupWizard.isPrimaryUser
+    // GSI requires an unlocked bootloader; never show "Disable OEM Unlocking"
+    val disableOemUnlockingVisible = false
 
     init {
         FinishActions
-- 
2.52.0

